eurovision/notebooks/story.ipynb

```diff
@@ -2090,7 +2039,269 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## NOTE: Slovakia only has one final entry in our dataset, in 1998.\n",
+    "## So, when the model is 'trained' on all data excluding 1998, Slovakia doesn't get a rescaled_rank as there is no data to work with.\n",
+    "## The 1998 predictions therefore don't include Slovakia (the Spearman coefficient is calculated based on all other entrants in 1998).\n",
+    "\n",
+    "all_years = df['year'].unique()\n",
+    "\n",
+    "def get_spearman_baseline_without(year):\n",
+    "    rescaled_ranks = train_baseline(excluded_years=[year])\n",
+    "    predictions_baseline = predict_baseline(rescaled_ranks, year)\n",
+    "    actual = get_actual_ranks(year)\n",
+    "    # Remove Slovakia from 1998\n",
+    "    if year == 1998:\n",
+    "        predictions_baseline = predictions_baseline.query('to_code2 != \"SK\"')\n",
+    "        actual = actual.query('to_code2 != \"SK\"')\n",
+    "    return(get_spearman(predictions_baseline, actual))\n",
+    "    \n",
+    "spearmans = [get_spearman_baseline_without(year) for year in all_years]\n",
+    "\n",
+    "print(f'Mean Spearman coefficient across {len(all_years)} years: {np.mean(spearmans)}')\n",
+    "\n",
+    "df_xv = pd.DataFrame({'year': all_years, 'spearman': spearmans})\n",
+    "fig, ax = plt.subplots(figsize=(6, 4))\n",
+    "ax.scatter(df_xv['year'], df_xv['spearman'], color='#6495ed')\n",
+    "ax.set(xlabel='Year', ylabel='Spearman coefficient', title='Cross-validation: baseline method')\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "It turns out that *on average*, the historical results of a country at Eurovision are a half-decent indicator of how well they will perform in any given year.\n",
+    "Of course, this is hardly a surprise.\n",
+    "Interestingly, we can pick out years with lower Spearman correlations as being 'anomalies' in this regard, where an underdog won.\n",
+    "The year with the lowest value (that's 2002) featured Latvia as the winner, and perhaps even more surprisingly, the UK finished third ([Jessica Garlick with *'Come Back'*](https://www.youtube.com/watch?v=2NSR5rSk0Cs)).\n",
+    "When we write 'surprisingly', this is emphatically *not* us inserting our personal opinion: the baseline model in fact really dislikes the UK, assigning it the third-lowest average rescaled rank of all countries.\n",
+    "Thus, any year where the UK performs well is a surprise to our model!\n",
+    "\n",
+    "Having looked at how good our model is at reproducing its own data, let's see what happens when we predict the 2023 results (now including all years' worth of data in our training)."
```

Comment:
Just to be absolutely clear, when you say "all year's worth of data" I think you're not including 2023 data, so it maybe worth making this explicit to avoid any confusion ("all years 1998 - 2022 inclusive" or something like that).

Reply:
It now reads, 'now including all data from 1998 to 2022, inclusive, in our training'.